oci/private/push.bzl

```diff
@@ -124,19 +141,27 @@ _attrs = {
         """,
         allow_single_file = [".txt"],
     ),
+    "_crane": attr.label(
+        cfg = _transition_to_target,
```

Comment:
Shouldn't this simply be `cfg = "exec"`?  What's the reason to transition oci_push target platform?

Reply:
Using cfg = “exec” would give you binaries suitable for use in ctx.actions.run(). But that doesn’t necessarily give you binaries that can be launched on the target, which is the platform for which oci_push() builds its launcher script.

Reply:
Hmm, perhaps i'm wrong, doesn't bazel use `exec` (aka host) configuration when running `bazel run :target`?

Reply:
No. That only holds if the first platform you register happens to be equal to your host platform.

If you run Bazel on a Mac, and you register Linux followed by macOS, then Bazel will try to execute as many actions as possible on Linux.

Reply:
^that.  Eg, when using bazel on mac with a linux RBE cluster (ie: `bazel run :push --platforms=mac --extra_execution_platforms=linux`), the `expand_template` action will have target=mac and exec=linux.  We need to find the crane/yq binaries _for mac_, because we're using their paths in our _output_ and not actually invoking these tools as part of the expand_template action itself.